File: group-anagrams/group-anagrams.py
class Solution:
    def groupAnagrams(self, strs):
        anagrams_map = {}

        for s in strs:
            sorted_s = ''.join(sorted(s))
            if not sorted_s in anagrams_map:
                anagrams_map[sorted_s] = [s]
            else:
                anagrams_map[sorted_s].append(s)

        return list(anagrams_map.values())

s = Solution()
print(s.groupAnagrams(["eat","tea","tan","ate","nat","bat"]))

File: group-anagrams/group-anagrams.js
var groupAnagrams = function (strs) {
  const anagramsMap = {}

  for (const s of strs) {
    const sorted_s = s.split('').sort().join('')
    if (anagramsMap[sorted_s]) {
      anagramsMap[sorted_s].push(s)
    } else {
      anagramsMap[sorted_s] = [s]
    }
  }

  return Object.values(anagramsMap)
}

console.log(groupAnagrams(["eat","tea","tan","ate","nat","bat"]))
